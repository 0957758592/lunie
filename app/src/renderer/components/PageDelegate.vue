<template lang="pug">
.page.page-candidates
  page-header(title='Delegate Atoms')
    countdown-string(date='2017-07-10')
    btn(theme='cosmos' type='link' to='/' icon="angle-left" value='Change Candidates')
  .candidates
    .candidates-container
      .delegation-form(v-for='c in filteredCandidates', key='c.id')
        card-candidate(:candidate='c')
        form-group
          label Delegate how many atoms?
          field(
            theme="cosmos"
            type="text"
            placeholder="Atoms"
            v-model="getShoppingCartItem(c.id).atoms")
</template>

<script>
import { mapGetters } from 'vuex'
import Btn from '@nylira/vue-button'
import CountdownString from './CountdownString'
import CardCandidate from './CardCandidate'
import Field from '@nylira/vue-input'
import FormGroup from './FormGroup'
import FormStruct from './FormStruct'
import PageHeader from './PageHeader'
export default {
  name: 'page-candidates',
  components: {
    CardCandidate,
    Btn,
    CountdownString,
    Field,
    FormGroup,
    FormStruct,
    PageHeader
  },
  computed: {
    ...mapGetters(['shoppingCart', 'candidates']),
    filteredCandidates () {
      let ids = this.shoppingCart.map(c => c.candidateId)
      return this.candidates.filter(c => ids.includes(c.id))
    }
  },
  methods: {
    getShoppingCartItem (candidateId) {
      return this.shoppingCart.find(c => c.candidateId === candidateId)
    }
  }
}
</script>
<style lang="stylus">
@import '../styles/variables.styl'

.delegation-form
  border 2px solid bc-dim
  margin-top 1rem
  border-radius 0.25rem
  padding 1rem - 2*px
  &:last-of-type
    margin-bottom 1rem

  .ni-form-group
    display flex
    margin -0.5rem -1rem -1rem
    label
      flex 1
    .ni-field
      flex 1
      width inherit

</style>

