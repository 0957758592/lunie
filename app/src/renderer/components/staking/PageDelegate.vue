<template lang="pug">
tm-page(:title="delegateType + ' Profile'")
  div(slot="menu"): tool-bar

  tm-part(:title="delegateType + ' Info'")
    tm-list-item(dt='Moniker' :dd='delegate.moniker')
    tm-list-item(dt='Website' :dd='delegate.website')
    tm-list-item(dt='Address' :dd='delegate.address')
    tm-list-item(dt='Public Key' :dd='delegate.pub_key.data')

  tm-part(:title="delegateType + ' Description'")
    text-block(:content="delegate.details || 'No description available.'")

  tm-part(:title="delegateType + ' Details'" v-if="isValidator")
    tm-list-item(dt="Total Vote Power" :dd="delegate.voting_power")
</template>

<script>
import { mapGetters } from "vuex"
import { TmListItem, TmPage, TmPart } from "@tendermint-ui"
import TextBlock from "common/TextBlock"
import ToolBar from "common/VrToolBar"
export default {
  name: "page-delegate",
  components: {
    TmListItem,
    TmPage,
    TmPart,
    TextBlock,
    ToolBar
  },
  computed: {
    ...mapGetters(["delegates"]),
    delegate() {
      if (this.delegates.delegates && this.$route.params.delegate) {
        return this.delegates.delegates.find(
          v => v.id === this.$route.params.delegate
        )
      } else {
        return {
          pub_key: {},
          voting_power: 0,
          owner: {}
        }
      }
    },
    isValidator() {
      if (this.delegate) {
        return this.delegate.voting_power > 0
      } else {
        return false
      }
    },
    delegateType() {
      return this.isValidator ? "Validator" : "Candidate"
    }
  }
}
</script>
