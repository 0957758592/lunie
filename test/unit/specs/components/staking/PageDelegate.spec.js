import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import PageDelegate from 'renderer/components/staking/PageDelegate'

const localVue = createLocalVue()
localVue.use(Vuex)

const $route = {
  params: {
    delegate: '1a2b3c'
  }
}

describe('PageDelegate', () => {
  let wrapper
  let store = new Vuex.Store({
    getters: {
      config: () => ({ desktop: false }),
      delegates: () => ({
        delegates: [
          {
            id: '1a2b3c',
            moniker: 'JB',
            website: 'https://the.zone',
            voting_power: 1000,
            owner: {
              addr: 'helloaddr'
            },
            pub_key: {
              data: '123pubkeyforme'
            }
          },
          {
            id: 'd4e5f6'
          }
        ]
      })
    }
  })

  beforeEach(() => {
    wrapper = mount(PageDelegate, {
      localVue,
      store,
      mocks: {
        $route
      }
    })

    wrapper.update()
  })

  it('has the expected html structure', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it('should return one delegate based on route params', () => {
    expect(wrapper.vm.delegate.id).toEqual('1a2b3c')
  })
})
