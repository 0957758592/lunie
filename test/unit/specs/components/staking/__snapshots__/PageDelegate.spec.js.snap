// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDelegate has the expected html structure 1`] = `
<div
  class="ni-page"
>
  <header
    class="ni-page-header"
    icon="storage"
  >
    <div
      class="ni-page-header-container"
    >
      <div
        class="ni-page-header-text"
      >
        <div
          class="ni-page-header-title"
        >
          <h2>
            Delegate
          </h2>
        </div>
        <div
          class="ni-page-header-subtitle"
        >
          <h3>
            
          </h3>
        </div>
      </div>
      <menu
        class="ni-page-header-menu"
      >
        <div>
          <div>
            <div
              class="ni-tool-bar"
            >
              <div
                class="ni-tool-bar-container"
              >
                <div
                  class="main"
                >
                  <a>
                    <i
                      class="material-icons"
                    >
                      add
                    </i>
                    <div
                      class="label"
                    >
                      Add
                    </div>
                  </a>
                </div>
                <a
                  class="back"
                >
                  <i
                    class="material-icons"
                  >
                    arrow_back
                  </i>
                  <div
                    class="label"
                  >
                    Back
                  </div>
                </a>
                <a
                  class="help"
                >
                  <i
                    class="material-icons"
                  >
                    help_outline
                  </i>
                  <div
                    class="label"
                  >
                    Help
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </menu>
    </div>
  </header>
  <main
    class="ni-page-main"
  >
    <h3>
      
    </h3>
    <section
      class="ni-part"
    >
      <div
        class="ni-part-container"
      >
        <header
          class="ni-part-header"
        >
          <div
            class="ni-part-header-container"
          >
            <div
              class="ni-part-title h5"
            >
              Delegate Details
            </div>
            <menu
              class="ni-part-menu"
            />
          </div>
        </header>
        <main
          class="ni-part-main"
        >
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Public Key
                </div>
                <div
                  class="ni-li-dd"
                >
                  pubkeyX
                </div>
              </div>
            </div>
          </div>
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Country
                </div>
                <div
                  class="ni-li-dd"
                >
                  United States
                </div>
              </div>
            </div>
          </div>
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Start Date
                </div>
                <div
                  class="ni-li-dd"
                >
                  n/a
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    </section>
    <section
      class="ni-part"
    >
      <div
        class="ni-part-container"
      >
        <header
          class="ni-part-header"
        >
          <div
            class="ni-part-header-container"
          >
            <div
              class="ni-part-title h5"
            >
              Delegate Description
            </div>
            <menu
              class="ni-part-menu"
            />
          </div>
        </header>
        <main
          class="ni-part-main"
        >
          <div
            class="text-block"
          >
            <pre>
              
            </pre>
          </div>
        </main>
      </div>
    </section>
    <section
      class="ni-part"
    >
      <div
        class="ni-part-container"
      >
        <header
          class="ni-part-header"
        >
          <div
            class="ni-part-header-container"
          >
            <div
              class="ni-part-title h5"
            >
              Delegate Stake
            </div>
            <menu
              class="ni-part-menu"
            />
          </div>
        </header>
        <main
          class="ni-part-main"
        >
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Voting Power
                </div>
                <div
                  class="ni-li-dd"
                >
                  10000 ATOM
                </div>
              </div>
            </div>
          </div>
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Bonded Atoms
                </div>
                <div
                  class="ni-li-dd"
                >
                  5000 ATOM
                </div>
              </div>
            </div>
          </div>
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Commission
                </div>
                <div
                  class="ni-li-dd"
                >
                  NaN%
                </div>
              </div>
            </div>
          </div>
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Max Commission
                </div>
                <div
                  class="ni-li-dd"
                >
                  NaN%
                </div>
              </div>
            </div>
          </div>
          <div
            class="ni-li"
          >
            <div
              class="ni-li-container"
            >
              <div
                class="ni-li-thumb"
              >
                <!---->
              </div>
              <div
                class="ni-li-dl"
              >
                <div
                  class="ni-li-dt"
                >
                  Max Commission Increase
                </div>
                <div
                  class="ni-li-dd"
                >
                  NaN%
                </div>
              </div>
            </div>
          </div>
        </main>
      </div>
    </section>
    <!---->
  </main>
</div>
`;
